compiler/src/Language/Bond/Codegen/Java/Util.hs

```diff
@@ -62,6 +63,21 @@ isPrimitiveBondType :: Type -> Bool
 isPrimitiveBondType (BT_UserDefined Enum {..} _) = True
 isPrimitiveBondType t = isPrimitiveNonEnumBondType t
 
+-- returns whether the given type maps to a Java primitive
+isJavaPrimitive :: Type -> Bool
+isJavaPrimitive BT_Int8 = True
+isJavaPrimitive BT_Int16 = True
+isJavaPrimitive BT_Int32 = True
+isJavaPrimitive BT_Int64 = True
+isJavaPrimitive BT_UInt8 = True
+isJavaPrimitive BT_UInt16 = True
+isJavaPrimitive BT_UInt32 = True
+isJavaPrimitive BT_UInt64 = True
+isJavaPrimitive BT_Float = True
+isJavaPrimitive BT_Double = True
+isJavaPrimitive BT_Bool = True
+isJavaPrimitive _ = False
```

Comment:
This returns false for (BT_UserDefined Alias), so the generated code will be wrong when a Java primitive type is aliased. Please refer to isPrimitiveNonEnumBondType for solution and I suggest you add a test case with an aliased primitive type.